### DISCUSSION TITLE
Send jsonl from Home Assistant: Doc unclear

### DISCUSSION THREAD

**Discussion by netmikey** [HASwitchPlate/openHASP#192] [Q&A] [answered]
Hi,

I'm in the process of setting up my first openHASP device, so I'm new and closely following the documentation. I'd like to keep all my configuration files in one place (preferably the HA config folder that's being backed up), so I'd like HA to send the configuration to the openHASP device. There is a section about this in the documentation. It says

> You can save your jsonl files in Home Assistant too. The custom component can send the contents of the file when the plate connects.

Perfect! However, the doc then goes on explaining how to enable read access for HomeAssistant on some "external directory" named `/config/openhasp`. I'm missing some info here. I'd expect the doc to tell me:

- where to put the jsonl files within Home Assistant (are they supposed to go in `/config/openhasp`?),
- whether this location is configurable or not (looks like it isn't?), and 
- how the integration choses the right jsonl file for the right plate (I'd guess the filename must match the plate name?).

A bit more context and info in this part of the documentation would be very welcome :-)

Best regards,
Mike

**Comment by nagyrobi** [comment, marked as answer]
Doc is unclear indeed because that part of the component is under develpoment. Answers to your questions:
- `/config/openhasp` directory is just an example. You can create any directory name in the location where your main `configuration.yaml` resides (that's `/config/`).
- when your plate is discovered, in the popup window of the configuration flow there's a field where you can enter the path for the jsonl file you want for that plate. Like `/config/openhasp/plate35.jsonl`.

Develpoment is still ongoing, in the future this may change. Feel free to ask if anything unclear.

Docs will be finalized after development of a certain feature will not change fundamentally, this is not yet set for this question.

**Reply by netmikey** [reply]
Oh, this explains that, thanks for the quick reply! :-)

So what if I haven't entered anything in this first initial text box, (how) can I set a jsonl file location _after_ the plate has been added initially?

**Reply by fvanroie** [reply]
Sure, go to Configuration > Integrations > HASP-Open Hardware Edition> *PlatName* > Configure

![image](https://user-images.githubusercontent.com/15969459/127679422-1cac2b7f-1f5b-4714-9e18-e9143032805f.png)

So, basically any path that is readable by HomeAssistant can be configured.

**Reply by nagyrobi** [reply]
You may need to delete it, reboot/restart, and rediscover it.

**Reply by netmikey** [reply]
Thanks, @fvanroie found it and it works.

**Reply by nagyrobi** [reply]
This is clarified a bit in the docs for 0.6.1.
